File: src/main/java/de/bnder/taskmanager/commands/Language.java
package de.bnder.taskmanager.commands;

import de.bnder.taskmanager.main.Command;
import de.bnder.taskmanager.main.Main;
import de.bnder.taskmanager.slashcommands.UpdateGuildSlashCommands;
import de.bnder.taskmanager.utils.Localizations;
import de.bnder.taskmanager.utils.MessageSender;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.Member;
import net.dv8tion.jda.api.entities.Role;
import net.dv8tion.jda.api.entities.TextChannel;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import org.jsoup.Connection;

import java.awt.*;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class Language implements Command {

    public static ArrayList<Locale> validLangCodes = new ArrayList<Locale>() {{
        add(Locale.GERMAN);
        add(Locale.ENGLISH);
        add(new Locale("bg"));
        add(Locale.FRENCH);
        add(new Locale("ru"));
        add(new Locale("pl"));
        add(new Locale("tr"));
        add(new Locale("pt"));
    }};

    @Override
    public void action(String[] args, String messageContentRaw, Member commandExecutor, TextChannel textChannel, Guild guild, java.util.List<Member> mentionedMembers, java.util.List<Role> mentionedRoles, List<TextChannel> mentionedChannels, SlashCommandEvent slashCommandEvent) throws IOException {
        if (args.length == 0) {
            final Locale langCode = Localizations.getGuildLanguage(guild);
            final String embedTitle = Localizations.getString("language_message_title", langCode);
            MessageSender.send(embedTitle, Localizations.getString("language_message_invalid_params", langCode), textChannel, Color.red, langCode, slashCommandEvent);
        } else if (args.length == 1) {
            final Locale language = Locale.forLanguageTag(args[0].toLowerCase());
            if (validLangCodes.contains(language)) {
                if (commandExecutor.hasPermission(Permission.ADMINISTRATOR)) {
                    final Connection.Response res = Main.tmbAPI("server/language/" + guild.getId(), commandExecutor.getId(), Connection.Method.POST).data("language", language.getLanguage()).execute();
                    final String embedTitle = Localizations.getString("language_message_title", language);
                    if (res.statusCode() == 200) {
                        UpdateGuildSlashCommands.update(guild);
                        MessageSender.send(embedTitle, Localizations.getString("language_changed", language), textChannel, Color.green, language, slashCommandEvent);
                    } else {
                        MessageSender.send(embedTitle, Localizations.getString("request_unknown_error", language), textChannel, Color.red, language, slashCommandEvent);
                    }
                } else {
                    final Locale langCode = Localizations.getGuildLanguage(guild);
                    final String embedTitle = Localizations.getString("language_message_title", langCode);
                    MessageSender.send(embedTitle, Localizations.getString("need_to_be_serveradmin_or_have_admin_permissions", langCode), textChannel, Color.red, langCode, slashCommandEvent);
                }
            } else {
                final Locale langCode = Localizations.getGuildLanguage(guild);
                final String embedTitle = Localizations.getString("language_message_title", langCode);
                MessageSender.send(embedTitle, Localizations.getString("language_message_invalid_params", langCode), textChannel, Color.red, langCode, slashCommandEvent);
            }
        }
    }
}

File: src/main/java/de/bnder/taskmanager/commands/Language.kt
package de.bnder.taskmanager.commands

import com.eclipsesource.json.Json
import de.bnder.taskmanager.main.Command
import de.bnder.taskmanager.main.Main
import de.bnder.taskmanager.utils.Connection
import de.bnder.taskmanager.utils.Localizations
import de.bnder.taskmanager.utils.MessageSender
import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
import org.jsoup.Jsoup
import java.awt.Color
import java.io.IOException
import java.util.*

/*
 * Copyright (C) 2019 Jan Brinkmann
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
class Language : Command {
    @Throws(IOException::class)
    override fun action(args: Array<String>, event: GuildMessageReceivedEvent) {
        val guild = event.guild
        if (args.size == 0) {
            val langCode = Localizations.getGuildLanguage(guild)
            val embedTitle = Localizations.getString("language_message_title", langCode)
            MessageSender.send(embedTitle, Localizations.getString("language_message_invalid_params", langCode), event.message, Color.red)
        } else if (args.size == 1) {
            val language = args[0].toLowerCase()
            if (language.equals("de", ignoreCase = true) || language.equals("en", ignoreCase = true) || language.equals("bg", ignoreCase = true)
                    || language.equals("fr", ignoreCase = true) || language.equals("ru", ignoreCase = true) || language.equals("pl", ignoreCase = true)) {
                if (Objects.requireNonNull(event.member)!!.hasPermission(Permission.ADMINISTRATOR)) {
                    val jsonResponse = Jsoup.connect(Main.requestURL + "setLanguage.php?requestToken=" + Main.requestToken + "&serverID=" + Connection.encodeString(guild.id) + "&language=" + Connection.encodeString(language)).userAgent(Main.userAgent).execute().body()
                    val `object` = Json.parse(jsonResponse).asObject()
                    val statusCode = `object`.getInt("status_code", 900)
                    val langCode = Localizations.getGuildLanguage(event.guild)
                    val embedTitle = Localizations.getString("language_message_title", langCode)
                    if (statusCode == 200) {
                        MessageSender.send(embedTitle, Localizations.getString("sprache_geändert", langCode), event.message, Color.green)
                    } else {
                        MessageSender.send(embedTitle, Localizations.getString("abfrage_unbekannter_fehler", langCode), event.message, Color.red)
                    }
                } else {
                    val langCode = Localizations.getGuildLanguage(event.guild)
                    val embedTitle = Localizations.getString("language_message_title", langCode)
                    MessageSender.send(embedTitle, Localizations.getString("muss_serverbesitzer_oder_adminrechte_haben", langCode), event.message, Color.red)
                }
            } else {
                val langCode = Localizations.getGuildLanguage(event.guild)
                val embedTitle = Localizations.getString("language_message_title", langCode)
                MessageSender.send(embedTitle, Localizations.getString("language_message_invalid_params", langCode), event.message, Color.red)
            }
        }
    }
}